internal/input/orderbooks.go

```diff
@@ -0,0 +1,258 @@
+package input
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"sync"
+
+	ingestio "github.com/stellar/go/ingest/io"
+	"github.com/stellar/go/ingest/ledgerbackend"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/stellar-etl/internal/transform"
+)
+
+type OrderbookBatch struct {
+	BatchStart uint32
+	BatchEnd   uint32
+	Orderbooks map[uint32][]ingestio.Change
+}
+
+// OrderbookParser handles parsing orderbooks
+type OrderbookParser struct {
+	Events            [][]byte
+	Markets           [][]byte
+	SeenMarketHashes  map[uint64]bool
+	Offers            [][]byte
+	SeenOfferHashes   map[uint64]bool
+	Accounts          [][]byte
+	SeenAccountHashes map[uint64]bool
+	Logger            *log.Entry
+	Strict            bool
+}
+
+func (o *OrderbookParser) convertOffer(allConvertedOffers []transform.NormalizedOfferOutput, index int, offer ingestio.Change, seq uint32, wg *sync.WaitGroup) {
+	defer wg.Done()
+	transformed, err := transform.TransformOfferNormalized(offer, seq)
+	if err != nil {
+		errorMsg := fmt.Sprintf("error json marshalling offer #%d in ledger sequence number #%d", index, seq)
+		if o.Strict {
+			o.Logger.Fatal(errorMsg, err)
+		} else {
+			o.Logger.Warning(errorMsg, err)
+		}
+	} else {
+		allConvertedOffers[index] = transformed
+	}
+}
+
+func NewOrderbookParser(strictExport bool, logger *log.Entry) OrderbookParser {
+	return OrderbookParser{
+		Events:            make([][]byte, 0),
+		Markets:           make([][]byte, 0),
+		SeenMarketHashes:  make(map[uint64]bool),
+		Offers:            make([][]byte, 0),
+		SeenOfferHashes:   make(map[uint64]bool),
+		Accounts:          make([][]byte, 0),
+		SeenAccountHashes: make(map[uint64]bool),
+		Logger:            logger,
+		Strict:            strictExport,
+	}
+}
+
+func (o *OrderbookParser) parseOrderbook(orderbook []ingestio.Change, seq uint32) {
+	var group sync.WaitGroup
+	allConverted := make([]transform.NormalizedOfferOutput, len(orderbook))
+	for i, v := range orderbook {
+		group.Add(1)
+		go o.convertOffer(allConverted, i, v, seq, &group)
+	}
+
+	group.Wait()
+
+	for _, converted := range allConverted {
+		if _, exists := o.SeenMarketHashes[converted.Market.ID]; !exists {
+			o.SeenMarketHashes[converted.Market.ID] = true
+			marshalledMarket, err := json.Marshal(converted.Market)
+			if err != nil {
+				errorMsg := fmt.Sprintf("error json marshalling market for offer: %d", converted.Offer.HorizonID)
+				if o.Strict {
+					o.Logger.Fatal(errorMsg, err)
+				} else {
+					o.Logger.Warning(errorMsg, err)
+					continue
+				}
+			} else {
```

Comment:
I might be wrong, but I think you don't need this in an `else` statement, since `Fatal` or `continue` would both move to the next iteration and not append the `marshalledMarket`.

This applies to a few other `else` clauses below.